Automattic/wp-calypso
# Framework: Refactor last `create-react-class` instances to HoCs and remove it

#### Changes proposed in this Pull Request

With recent `create-react-class` refactors, we only have 2 remaining instances of `create-react-class` in the Happychat components.

This PR refactors the mixins that are used there to high-order components. It also updates the usages of those components to no longer use `create-react-class` but use `React.FC` / `React.Component` and the high order components instead. We finally remove the no longer used mixins and the `create-react-class` dependency completely 🎉.

#### Testing instructions

* Verify that removed code (mixins and `create-react-class`) isn't used anywhere anymore.
* Make sure you have HE permissions (you need to be added to at least one happiness queue).
* Go to https://hud-staging.happychat.io/ and enable yourself as an agent, adding at least 1 available slot.
* Open any Calypso page that has the help button (for example, `/settings/general/:site` for a paid site)
* Click the Help button on the bottom right, and then initiate a chat.
* Go back and forth sending messages from the HUD and from Calypso, and make sure auto-scrolling and scroll bleed continue to work as they do in production. Try refreshing Calypso, make sure everything chat-wise still works as before.
* Keep an eye on the console, verify there are no new errors / warnings introduced by these changes.
* Verify tests still pass: `yarn run test-client client/components/happychat`
